src/refactor_nrepl/find/find_symbol.clj

```diff
@@ -103,41 +102,42 @@
          (str/join "\n")
          str/trim)))
 
-(defn- find-symbol-in-file [fully-qualified-name ignore-errors ^File file]
+(defn- find-symbol-in-file [fully-qualified-name ignore-errors referred-syms ^File file]
   (let [file-content (slurp file)
         locs (try (->> (ana/ns-ast file-content)
                        (find-symbol-in-ast fully-qualified-name)
                        (filter :line-beg))
                   (catch Exception e
                     (when-not ignore-errors
                       (throw e))))
-        locs (concat locs
-                     (some->
-                      (libspecs/referred-syms-by-file&fullname)
-                      (get-in [:clj (str file) fully-qualified-name])
-                      meta
-                      ((fn [{:keys [line column end-line end-column]}]
-                         (list {:line-beg line
-                                :line-end end-line
-                                :col-beg column
-                                :col-end end-column})))))
+        locs (into
+              locs (some->
+                    referred-syms
+                    (get-in [:clj (str file) fully-qualified-name])
+                    meta
+                    ((fn [{:keys [line column end-line end-column]}]
+                       (list {:line-beg line
+                              :line-end end-line
+                              :col-beg column
+                              :col-end end-column})))))
         gather (fn [info]
                  (merge info
                         {:file (.getCanonicalPath file)
                          :name fully-qualified-name
                          :match (match file-content
-                                  (:line-beg info)
-                                  (:line-end  info))}))]
+                                       (:line-beg info)
+                                       (:line-end  info))}))]
     (map gather locs)))
 
 (defn- find-global-symbol [file ns var-name ignore-errors]
   (let [namespace (or ns (core/ns-from-string (slurp file)))
         fully-qualified-name (if (= namespace "clojure.core")
                                var-name
-                               (str/join "/" [namespace var-name]))]
+                               (str/join "/" [namespace var-name]))
+        referred-syms (libspecs/referred-syms-by-file&fullname)]
```

Comment:
calculating `referred-syms` once and passing result to each call of `find-symbol-in-file fully-qualified-name`